_posts/2022-03-25-deep-learning-for-routing-problems.md

```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+title: Recent Advances in Deep Learning for Routing Problems
+tags: [deep-learning, graph-neural-networks, combinatorial-optimization, travelling-salesperson-problem]
+authors: Joshi, Chaitanya K.; Anand, Rishabh
+---
+
+**TL;DR** Developing neural network-driven solvers for combinatorial optimization problems such as the Travelling Salesperson Problem have seen a surge of academic interest recently. This blogpost presents a **Neural Combinatorial Optimization** pipeline that unifies several recently proposed model architectures and learning paradigms into one single framework. Through the lens of the pipeline, we analyze recent advances in deep learning for routing problems and provide new directions to stimulate future research towards practical impact.
+
+{: class="table-of-content"}
+* TOC
+{:toc}
+
+---
+
+## Background on Combinatorial Optimization Problems
+
+**Combinatorial Optimization** is a practical field in the intersection of mathematics and computer science that aims to solve constrained optimization problems which are NP-Hard. **NP-Hard problems** are challenging as exhaustively searching for their solutions is beyond the limits of modern computers. It is impossible to solve NP-Hard problems optimally at large scales. 
+
+**Why should we care?** Because robust and reliable approximation algorithms to popular problems have immense practical applications and are the backbone of modern industries. For example, the **Travelling Salesperson Problem** (TSP) is the most popular Combinatorial Optimization Problems (COPs) and comes up in applications as diverse as logistics and scheduling to genomics and systems biology. 
+
+> The Travelling Salesperson Problem is so famous, or notorious, that it even has an [xkcd comic](https://xkcd.com/399/) dedicated to it!
+
+### TSP and Routing Problems
+
+TSP is also a classic example of a **Routing Problem** -- Routing Problems are a class of COPs that require a sequence of nodes (e.g., cities) or edges (e.g., roads between cities) to be traversed in a specific order while fulfilling a set of constraints or optimising a set of variables. TSP requires a set of edges to be traversed in an order that ensures all nodes are visited exactly once. In the algorithmic sense, the optimal "tour" for our salesperson is a sequence of selected edges that provides the minimal distance or time taken over a Hamiltonian cycle, see Figure 1 for an illustration.
+
+<figure><center>
+  <img src="{{ site.url }}/public/images/2021-12-01-deep-learning-for-routing-problems/tsp-gif.gif" width="60%"/>
```

Comment:
When you update the images asset directory to `2022-03-05-deep-learning-for-routing-problems`, you will need to propagate these changes in your post as well (sorry about that)